src/QUICStream.ts

```diff
@@ -475,7 +475,8 @@ class QUICStream implements ReadableWritablePair<Uint8Array, Uint8Array> {
       const [recvLength] = result;
       if (recvLength > 0) {
         this.readableController.enqueue(
-          this.readableChunk!.subarray(0, recvLength),
+          // Making a copy to be enqueued
+          Buffer.from(this.readableChunk!.slice(0, recvLength)),
```

Comment:
I thought that this being synchronous would mean it would be fine.

Reply:
It's not a race problem.  when we enqueue a subarray its a view of the underlying buffer's memory. So when we enqueue the 2nd message they're referencing the same memory. It means that the data in the buffer is only valid until the next read.

I've applied this patch which is to just copy the slice we want and enqueue that. I think that's safer from a usage perspective. But if we loos about 300 ops/s on my laptop.

If we want the performance route then we change this back. while using it we just make a copy of the data if we want to hold onto after further `read()`s.

Reply:
I'm confused because I thought the "next read" would have just overwritten the readable chunk, so how could clobbering even happen?

Reply:
For example:

```
        result = this.connection.conn.streamRecv(
          this.streamId,
          this.readableChunk!,
        );
```

Would overwrite the `this.readableChunk!`, so if it's not a race condition, how can any clobbering happen?

Reply:
This is new issue https://github.com/MatrixAI/js-quic/issues/65

Reply:
Was there an answer to this questions?

Reply:
When we passed the buffer to the readable stream queue we pass by reference since it's an object. When consuming the readable stream we get the object can use it. In most cases we'd read, do something with it and be done with it before the next read and the next time the buffer changed.

The problem was that in one case we'd read and keep the buffer somewhere and then read again. Both reads write to and return the same object.

So doing something like consuming the stream by pushing each message into an array. The array will be filled with multiple references to the same buffer that had just been overwritten at each read.